### ISSUE TITLE
Don't temporary store (encrypted) passwords in confirmation

### ISSUE THREAD

**Issue by joostdecock** [freesewing/backend#1]
Because the mongoose bcrypt plugin hashes the password on save, we cannot create an account from a pre-hashed password, as it will be hashed twice.

To sidestep that issue, we are currently storing the password in an encrypted field in a confirmation.
However, it would be better to never store the password (even if it's encrypted) in it's non-hashed form.

There are some possible solutions outlined in this ticket: proswdev/mongoose-bcrypt#15

Duplicated at https://github.com/freesewing/website/issues/75 for exposure

**Comment by joostdecock** [comment]
Here's how this could be handled:

When saving the document, check whether it's a new one (using `document.isNew()`)  
However, `isNew()` is only available in pre hooks, not in post hooks. So we would have to store the value somewhere. Like this:

```
schema.pre('save', function (next) {
    this.wasNew = this.isNew;
    this.passHash = this.password;
    next();
});

schema.post('save', function () {
    if (this.wasNew) this.password = this.passHash;
});
```
By not defining `wasNew` or `passHash` in the scheme, it is my understanding that they won't be saved when the document is actually created.
This would work provided that:

 - We can make sure our `pre` hooks runs after the bcrypt `pre` hook
 - We configure a password field in confirmations that gets encrypted with bcrypt

**Comment by joostdecock** [comment]
So in the end I rewrote the signup flow so that an account is created upon signup.  
My initial reasoning for not doing this is that we should not store personal data (like the email address) prior to receiving GDPR consent from the user. However:

 - Storing the data elsewhere is still storing it
 - Without confirmation that this is actually the email address of the user, it's arguably not personal data
 - Account creation (activation) is only available after giving consent

Long story short, it's been dealt with.
